app/services/UpdateAddressResponse.scala

```diff
@@ -22,18 +22,16 @@ import error.GenericErrors
 import play.api.i18n.Messages
 import play.api.mvc.Result
 import uk.gov.hmrc.http.HttpResponse
-import util.LocalPartialRetriever
 
 sealed trait UpdateAddressResponse {
-  def response(successResponseBlock: () => Result)(
+  def response(genericErrors: GenericErrors, successResponseBlock: () => Result)(
     implicit request: UserRequest[_],
     configDecorator: ConfigDecorator,
-    partialRetriever: LocalPartialRetriever,
     messages: Messages): Result =
     this match {
-      case UpdateAddressBadRequestResponse    => GenericErrors.badRequest
-      case UpdateAddressUnexpectedResponse(_) => GenericErrors.internalServerError
-      case UpdateAddressErrorResponse(_)      => GenericErrors.internalServerError
+      case UpdateAddressBadRequestResponse    => genericErrors.badRequest
+      case UpdateAddressUnexpectedResponse(_) => genericErrors.internalServerError
+      case UpdateAddressErrorResponse(_)      => genericErrors.internalServerError
```

Comment:
Since these 2 cases do the same thing we can merge them together like this
case UpdateAddressUnexpectedResponse(_) || UpdateAddressErrorResponse(_) => genericErrors.internalServerError

Reply:
Done